docker/default-config/rabbitmq-definitions.json

```diff
@@ -3,7 +3,7 @@
   "users": [
     {
       "name": "harvest",
-      "password_hash": "<ADD PASSWORD HASH HERE> (The Python script available at the https://stackoverflow.com/questions/41306350/how-to-generate-password-hash-for-rabbitmq-management-http-api/53016240#53016240 can be used to generate a password hash for a new password hash)",
+      "password_hash": "d/hQxz0npBVirMPzW821BijSvzlozRiWAynAzgPzyXNhpqa4",
```

Comment:
Is this ok? 😦🤔

Reply:
@nutjob4life, I had the same question. 

This is the expectation of issue "Simplify the readme and procedure to start the registry with docker-compose#241"

Copying @tloubrieu-jpl and @jordanpadams too.

It says,
Use the **default user/password** for rabbitmq so that everything can be started without update.

This is only for quick start and I specially mentioned in the README that,

"**PLEASE DO NOT** use the quick start with default configurations in any environment other than your local machine. Make sure to replace the default passwords with your own passwords as explained in the following section:
**Steps to configure registry components to be executed with docker compose** to ensure security."

Reply:
Ok thanks! 😅

Reply:
👍 we may get flagged for this at some point, but I am OK with it for now.

Reply:
@nutjob4life I like this idea. any way we can just change that to the default to be very very clear? this will help when we eventually get scanned. if they see something that doesn't look like nonsense, we will get dinged

Reply:
@jordanpadams, @nutjob4life and @tloubrieu-jpl ,

Another suggestion, as follows.

Keep it in the code (`docker/default-config/rabbitmq-definitions.json`) as follows:

```
{
    "password_hash": "--INSECURE-HASH-FOR-TESTING-PURPOSES--"
}
```

Then, add an additional step in the README file as follows.

2) Open `docker/default-config/rabbitmq-definitions.json` file and update the
`"password_hash": "--INSECURE-HASH-FOR-TESTING-PURPOSES--"` with `"password_hash": "d/hQxz0npBVirMPzW821BijSvzlozRiWAynAzgPzyXNhpqa4"`


That way, this will not be a problem during the scans. However, on the other hand it will add one additional step to the quick start guide. Also, please note that most of the configuration files used by scalable Harvest components have the default password of RabbbitMQ.

What do you think?

Reply:
Why don't you use Docker secrets https://docs.docker.com/engine/swarm/secrets/

Reply:
Isn't this just for testing? 🤔

If it's just for testing, it doesn't matter what the secret password hash is. Tests run, done. Who cares 🤷‍♀️

If it's for operations or on a persistently available system or service, then yes, we need another step in the README or use Docker Secrets or something.

Reply:
> Why don't you use Docker secrets https://docs.docker.com/engine/swarm/secrets/ It is available in "swarm" mode only, though.

@tdddblog, this JSON file is not a docker specific file. It is an initialization file used to initialize RabbitMQ (to create use accounts). If want to use docker secrets, then we have read docker secrets and rewrite the JSON file, that can be an overkill for local quick setup purpose. 

This is similar to having the default RabbitMQ password specified in the following list of files. Only difference is: in the JSON file we have it as a hash instead of in plain text.

https://github.com/NASA-PDS/registry-harvest-cli/blob/main/src/main/resources/conf/harvest-client.cfg
https://github.com/NASA-PDS/registry-harvest-service/blob/main/src/main/resources/conf/harvest-server.cfg
https://github.com/NASA-PDS/registry-crawler-service/blob/main/src/main/resources/conf/crawler-server.cfg

Reply:
Oh okay, so it's not like a secret key that's used to hash passwords. It _is the test password_, but in hashed form.

In that case, I say go ahead and use the `d/hQxz0npBVirMPzW821BijSvzlozRiWAynAzgPzyXNhpqa4` string and if any automated system complains we just suppress the warning.

Reply:
@ramesh-maddegoda can we have the clear default rabbitmq passowrd used (guest/guest) so that it is unambiguous that this is the default configuration. Or maybe we should not have any specific configuration for that  and just update the clients so that they use the default password until the user updates it.

All the specific configuration for having a secured rabbitmq should be detailed in the readme bu the default version coming out of github does not need to be secured. At least that is how I am seeing that.

Reply:
@tloubrieu-jpl, I tried that too. But RabbitMQ definition wants to have the password hashed with `rabbit_password_hashing_sha256` algorithm.  Also, as you suggested, we can use default RabbitMQ password used in all config files too.